sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S4426.html

```diff
@@ -1,7 +1,84 @@
-<p>When generating cryptograpic keys (or key pairs), it is important to use a key length that provides enough entropy against brute-force attacks. For
-the RSA algorithm the key should be at </p>
-<p> least 2048 bits long.</p>
-<p>This rule raises an issue when a RSA key-pair generator is initialized with too small a length parameter.</p>
+<p>Most of cryptographic systems require a sufficient key size to be robust against brute-force attacks.</p>
+<p><a href="https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar1.pdf">NIST recommendations</a> will be checked:</p>
+<table>
+  <tbody>
+    <tr>
+      <th>Use / Algorithm</th>
+      <th>DSA</th>
+      <th>RSA</th>
+      <th>ECDSA</th>
+      <th>DH</th>
+      <th>MQV</th>
+      <th>ECDH</th>
+      <th>ECMQV</th>
+      <th>Block Cipher</th>
+    </tr>
+    <tr>
+      <td>Digital Signature Generation </td>
+      <td> p ≥ 2048 AND q ≥ 224</td>
+      <td>n ≥ 2048</td>
+      <td>See table below </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x <br></td>
+      <td>Digital Signature Verification </td>
+      <td> p ≥ 2048 AND q ≥ 224</td>
+      <td>n ≥ 2048</td>
+      <td>See table below </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x <br></td>
+      <td>Key Agreement </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> p ≥ 2048 AND q ≥ 224 </td>
+      <td> p ≥ 2048 AND q ≥ 224 </td>
+      <td> See table below </td>
+      <td> See table below </td>
+      <td> x <br></td>
+      <td> Encryption and Decryption </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> x </td>
+      <td> AES-128, 192, 256<br><br>This rule will not raise issues for ciphers that are considered weak (no matter the key size) like
+      <code>DES</code>, <code>Blowfish</code>.<br><br>h3. Notation<br>* DSA ([Digital Signature Algorithm</td>
+      <td>https://en.wikipedia.org/wiki/Digital_Signature_Algorithm]): <code>p</code> is key length and <code>q</code> the modulus length<br>* EC
+      ([Elliptic-curve</td>
+      <td>https://en.wikipedia.org/wiki/Elliptic-curve_cryptography])<br></td>
+      <td></td>
+      <td>EC parameters</td>
```

Comment:
Changes in this file should be reverted, as the html generation is wrong here. What's more, new description does not fit current implementation anyways